### ISSUE TITLE
Training language model,can you help me?

### ISSUE THREAD

**Issue by genbei** [pytorch/fairseq#2497]
I used fairseq to train a language model, arch=transformer_lm, the data set is iwslt14, The trainingdata and validation data consist of 160k and 7k sentence pairs.And I used a GPU. I have been training for more than two days now, and the training is not over yet. loss and ppl have not decreased, even a little higher, and I think they tend to converge.
The first time I trained the language model, I don't know if he will stop training by himself, or if I think it will stop, or if my training parameters are wrong.
What should I do now, wait for him to finish by himself

`python train.py $DATA  --task language_modeling --arch transformer_lm \
  --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
  --lr-scheduler inverse_sqrt --warmup-init-lr 1e-07 --warmup-updates 4000 \
  --lr 0.0005 --min-lr 1e-09 \
  --dropout 0.1 --weight-decay 0.0 --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
  --max-tokens 4096  --tokens-per-sample 4096  --save-dir $SAVE --update-freq 16`

![3-15](https://user-images.githubusercontent.com/57440758/90606194-0d716980-e232-11ea-9654-729dbc3f1aa9.png)

**Comment by MrityunjoyS** [comment]
I think there is a parameter, [--max-epoch N], that you've to mention, else model will keep on running

**Comment by genbei** [comment]
> 我认为您必须提到一个参数[--max-epoch N]，否则模型将继续运行

`$ fairseq-train --task language_modeling \
  data-bin/wikitext-103 \
  --save-dir checkpoints/transformer_wikitext-103 \
  --arch transformer_lm --share-decoder-input-output-embed \
  --dropout 0.1 \
  --optimizer adam --adam-betas '(0.9, 0.98)' --weight-decay 0.01 --clip-norm 0.0 \
  --lr 0.0005 --lr-scheduler inverse_sqrt --warmup-updates 4000 --warmup-init-lr 1e-07 \
  --tokens-per-sample 512 --sample-break-mode none \
  --max-tokens 2048 --update-freq 16 \
  --fp16 \
  --max-update 50000`

Why doesn't the official script have this parameter --max-epoch N,
So according to the script, won't it stop automatically

**Comment by myleott** [comment]
You can either specify `--max-epoch` or `--max-update`. The README uses `--max-update 50000`, but that seems to be too large for your dataset, thus using `--max-epoch` may be easier to configure.

**Comment by genbei** [comment]
Thank you. I see




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "pytorch/fairseq"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年8月20日(星期四) 晚上10:38
收件人:&nbsp;"pytorch/fairseq"<fairseq@noreply.github.com&gt;;
抄送:&nbsp;"推杯换盏"<374481729@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [pytorch/fairseq] Training language model,can you help me? (#2497)





 
You can either specify --max-epoch or --max-update. The README uses --max-update 50000, but that seems to be too large for your dataset, thus using --max-epoch may be easier to configure.
